VultisigApp/VultisigApp/Model/FunctionCall/FunctionCallAddThorLP.swift

```diff
@@ -0,0 +1,524 @@
+    var addressFields: [String: String] {
+        get { ["pairedAddress": pairedAddress] }
+        set {
+            if let v = newValue["pairedAddress"] {
+                pairedAddress = v
+            }
+        }
+    }
+    
+    // MARK: Init
+    required init(tx: SendTransaction, functionCallViewModel: FunctionCallViewModel, vault: Vault) {
+        self.tx = tx
+        self.functionCallViewModel = functionCallViewModel
+        self.vault = vault
+        
+        prefillPairedAddress()
+        setupValidation()
+        loadInitialState()
+    }
+    
+    private func loadInitialState() {
+        fetchInboundAddressAndSetupApproval()
+        Task { @MainActor in
+            try? await Task.sleep(nanoseconds: 100_000_000) // 0.1s delay to let UI settle
+            await loadPools()
+        }
+    }
+    
+    // MARK: - Inbound address + approval
+    
+    private func fetchInboundAddressAndSetupApproval() {
+        Task { @MainActor in
+            let addresses = await ThorchainService.shared.fetchThorchainInboundAddress()
+            
+            if tx.coin.chain == .thorChain {
+                // For THORChain, we don't need an inbound address initially (it's set when pool is selected)
+                // The toAddress will be set when pool is selected
+
+                isApprovalRequired = false
+                approvePayload = nil
+                return
+            } else {
+                // Normal send path: need inbound address for L1/EVM chains.
+                let chainName = getInboundChainName(for: tx.coin.chain)
+                guard let inbound = addresses.first(where: { $0.chain.uppercased() == chainName.uppercased() }) else {
+                    return
+                }
+                
+                if inbound.halted || inbound.global_trading_paused || inbound.chain_trading_paused || inbound.chain_lp_actions_paused {
+                    return
+                }
+                
+                let destinationAddress: String
+                if tx.coin.shouldApprove {
+                    // ERC20 token (e.g., USDC) → approval to router
+                    destinationAddress = inbound.router ?? inbound.address
+
+                } else {
+                    // Native token (e.g., ETH) → direct to inbound address
+                    destinationAddress = inbound.address
+
+                }
+                
+                tx.toAddress = destinationAddress
+                
+                // ERC20 approval only for non-RUNE ERC20 tokens
+                isApprovalRequired = tx.coin.shouldApprove
+                if isApprovalRequired {
+                    if !tx.toAddress.isEmpty {
+                        let payload = ERC20ApprovePayload(amount: tx.amountInRaw, spender: tx.toAddress)
+                        self.approvePayload = payload
+
+                    } else {
+                        Task {
+                            for attempt in 0..<5 {
```

Comment:
I don't understand this logic , why you need to try 5 times just to set the ERC20ApprovePayload?